gensim/models/keyedvectors.py

```diff
@@ -311,11 +321,18 @@ def doesnt_match(self, words):
         """
         self.init_sims()
 
-        words = [word for word in words if word in self.vocab]  # filter out OOV words
-        logger.debug("using words %s" % words)
         if not words:
             raise ValueError("cannot select a word from an empty list")
-        vectors = vstack(self.syn0norm[self.vocab[word].index] for word in words).astype(REAL)
+        logger.debug("using words %s" % words)
+        vectors = []
+        for word in words:
```

Comment:
`words` was also the name of one of the params of the method, and the removed line was filtering out oov words.
Renamed/reworked things to make things a little clearer now.